static/app/utils/replays/replayReader.tsx

```diff
@@ -48,31 +49,48 @@ export default class ReplayReader {
    * The original `this._event.startTimestamp` and `this._event.endTimestamp`
    * are the same. It's because the root replay event is re-purposing the
    * `transaction` type, but it is not a real span occuring over time.
-   * So we need to figure out the real end time, in milliseconds.
+   * So we need to figure out the real start and end timestamps based on when
+   * first and last bits of data were collected. In milliseconds.
    */
-  private _getEndTimestampMS() {
+  private _getTimestampsMS = memoize(() => {
```

Comment:
memoize because it's got that sorting business and is called twice: from `getEvent` as well as `getRRWebEvents`, both of which are also memoized.

Could be slightly premature